pkg/driver/kubernetes_driver.go

```diff
@@ -0,0 +1,366 @@
+		"MASTER_URL":      "Kubernetes master endpoint",
+	}
+}
+
+// SetConfig sets Kubernetes driver configuration
+func (k *KubernetesDriver) SetConfig(settings map[string]string) {
+	k.setDefaults()
+	k.Namespace = settings["KUBE_NAMESPACE"]
+	k.ServiceAccountName = settings["SERVICE_ACCOUNT"]
+
+	var kubeconfig string
+	if kpath := settings["KUBE_CONFIG"]; kpath != "" {
+		kubeconfig = kpath
+	} else if home := homeDir(); home != "" {
+		kubeconfig = filepath.Join(home, ".kube", "config")
+	}
+
+	conf, err := clientcmd.BuildConfigFromFlags(settings["MASTER_URL"], kubeconfig)
+	if err != nil {
+		panic(err)
+	}
+	err = k.setClient(conf)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (k *KubernetesDriver) setDefaults() {
+	k.SkipCleanup = false
+	k.BackoffLimit = 0
+	k.ActiveDeadlineSeconds = 300
+	k.requiredCompletions = 1
+	k.deletionPolicy = metav1.DeletePropagationBackground
+}
+
+func (k *KubernetesDriver) setClient(conf *rest.Config) error {
+	coreClient, err := coreclientv1.NewForConfig(conf)
+	if err != nil {
+		return err
+	}
+	batchClient, err := batchclientv1.NewForConfig(conf)
+	if err != nil {
+		return err
+	}
+	k.jobs = batchClient.Jobs(k.Namespace)
+	k.secrets = coreClient.Secrets(k.Namespace)
+	k.pods = coreClient.Pods(k.Namespace)
+
+	return nil
+}
+
+// Run executes the operation inside of the invocation image
+func (k *KubernetesDriver) Run(op *driver.Operation) error {
+	if k.Namespace == "" {
+		return fmt.Errorf("KUBE_NAMESPACE is required")
+	}
+	labelMap := generateLabels(op)
+	meta := metav1.ObjectMeta{
+		Namespace:    k.Namespace,
+		GenerateName: generateNameTemplate(op),
+		Labels:       labelMap,
+	}
+	// Mount SA token if a non-zero value for ServiceAccountName has been specified
+	mountServiceAccountToken := k.ServiceAccountName != ""
+	job := &batchv1.Job{
+		ObjectMeta: meta,
+		Spec: batchv1.JobSpec{
+			ActiveDeadlineSeconds: &k.ActiveDeadlineSeconds,
+			Completions:           &k.requiredCompletions,
+			BackoffLimit:          &k.BackoffLimit,
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labelMap,
+				},
+				Spec: v1.PodSpec{
+					ServiceAccountName:           k.ServiceAccountName,
+					AutomountServiceAccountToken: &mountServiceAccountToken,
+					RestartPolicy:                v1.RestartPolicyNever,
+				},
+			},
+		},
+	}
+	container := v1.Container{
+		Name:    k8sContainerName,
+		Image:   op.Image,
+		Command: []string{"/cnab/app/run"},
+		Resources: v1.ResourceRequirements{
+			Limits: v1.ResourceList{
+				v1.ResourceCPU:    k.LimitCPU,
+				v1.ResourceMemory: k.LimitMemory,
+			},
+		},
+		ImagePullPolicy: v1.PullIfNotPresent,
+	}
+
+	if len(op.Environment) > 0 {
+		secret := &v1.Secret{
+			ObjectMeta: meta,
+			StringData: op.Environment,
+		}
+		secret.ObjectMeta.GenerateName += "env-"
+		envsecret, err := k.secrets.Create(secret)
+		if err != nil {
+			return err
+		}
+		// TODO: add to slice of resources awaiting cleanup instead, then return err
```

Comment:
Looks like a TODO was left over. Did you intend to address it in this PR?

Reply:
Actually I think this is fine. Removed the comment.